File: Source/Padutronics.Windows.Win32.Api/D2D1/ID2D1RadialGradientBrush.cs
using Padutronics.Windows.Win32.Api.DCommon;
using System.Runtime.InteropServices;

namespace Padutronics.Windows.Win32.Api.D2D1;

[Guid("2cd906ac-12e2-11dc-9fed-001143a055f9")]
public interface ID2D1RadialGradientBrush : ID2D1Brush
{
    D2D_POINT_2F GetCenter();
    D2D_POINT_2F GetGradientOriginOffset();
    void GetGradientStopCollection(out ID2D1GradientStopCollection gradientStopCollection);
    float GetRadiusX();
    float GetRadiusY();
    void SetCenter(D2D_POINT_2F center);
    void SetGradientOriginOffset(D2D_POINT_2F gradientOriginOffset);
    void SetRadiusX(float radiusX);
    void SetRadiusY(float radiusY);
}

File: Source/Padutronics.Windows.Win32.Api.CWrapper/D2D1/ID2D1RadialGradientBrush.cpp
#include "ID2D1RadialGradientBrush.hpp"

auto ID2D1RadialGradientBrush_GetCenter(ID2D1RadialGradientBrush& _this) -> D2D1_POINT_2F
{
    return _this.GetCenter();
}

auto ID2D1RadialGradientBrush_GetGradientOriginOffset(ID2D1RadialGradientBrush& _this) -> D2D1_POINT_2F
{
    return _this.GetGradientOriginOffset();
}

auto ID2D1RadialGradientBrush_GetRadiusX(ID2D1RadialGradientBrush& _this) -> FLOAT
{
    return _this.GetRadiusX();
}

auto ID2D1RadialGradientBrush_GetRadiusY(ID2D1RadialGradientBrush& _this) -> FLOAT
{
    return _this.GetRadiusY();
}